### PR TITLE
Enable popup submenu in Edge

### PR DESCRIPTION
**Pull request by OPersian**

### CODE REVIEW

File: video_xblock/static/js/player-context-menu.js [hunk 1]
```diff
@@ -40,10 +57,26 @@ domReady(function() {
     });
     docfrag.appendChild(ulSubMenu);
 
+    // Check conditions to be met for delegation of the popup submenu creation
+    var menuItemClicked = matchesSelector(target, "li.vjs-menu-item");
+    var noSubmenuClicked = !target.querySelector('.vjs-contextmenu-ui-submenu');
+    var menuItemsLabelsEqual = (labelElement === labelItem)
```
Comment (z4y4ts) [inline video_xblock/static/js/player-context-menu.js]: Trailing `;`

File: video_xblock/static/js/player-context-menu.js [hunk 2]
```diff
@@ -40,10 +57,26 @@ domReady(function() {
...
+
+    // Wrap into conditional statement to avoid unnecessary variables initialization
+    if (menuItemClicked && noSubmenuClicked) {
+      var labelLength = labelElement.length;
+      // Check if the last character is an escaped one (line feed) which is the case for Microsoft Edge
+      if (labelElement.charCodeAt(labelLength-1) === 10) {
```
Comment (z4y4ts) [inline video_xblock/static/js/player-context-menu.js]: Make `10` a constant

File: video_xblock/static/js/player-context-menu.js [hunk 3]
```diff
@@ -40,10 +57,26 @@ domReady(function() {
...
+        var labelElementSliced = [];
+        for (var i=0; i<labelLength-1; i++){
```
Comment (z4y4ts) [inline video_xblock/static/js/player-context-menu.js]: Can we use `forEach` here?

File: video_xblock/static/js/player-context-menu.js [hunk 4]
```diff
@@ -40,10 +57,26 @@ domReady(function() {
...
+          labelElementSliced += labelElement[i];
```
Comment (z4y4ts) [inline video_xblock/static/js/player-context-menu.js]: `+=` to an Array? How do you expect it to behave?

Comment (OPersian) [inline video_xblock/static/js/player-context-menu.js]: Sorry, remains from an old implementation; should be string

**Comment by z4y4ts** [comment]
Squash and merge

**Pull request merged**